zerver/lib/slack_data_to_zulip_data.py

```diff
@@ -543,6 +543,13 @@ def channel_message_to_zerver_message(constants: List[Any], channel: str,
     for json_name in json_names:
         messages = get_data_file(slack_data_dir + '/%s/%s' % (channel, json_name))
         for message in messages:
+
+            user = get_message_sending_user(message)
+            if not user:
```

Comment:
Does this mean, the `get_model_id` overcounts the number of messages?

Reply:
No, some of the messages have no user, example: https://github.com/rheaparekh/slack2zulip-data-converter/blob/master/slack-export/general/2018-02-23.json#L9-L83 (message generated by Slack Polling app)

Reply:
I think for those messages, the "sender" in the Slack UI probably appears as Slackbot, right?  Can you find that message in the UI and take a screenshot to confirm?

If that's what's going on, we want these messages to show up in Zulip -- we should just set their sender to be a new bot user added to the import called "slackbot".

Reply:
@timabbott the sender is the Slack Polling app which I had integrated.

Reply:
![Slack Polling app message](https://user-images.githubusercontent.com/25330892/36655106-b90a8670-1ae6-11e8-873b-c21fb9175898.png)

Reply:
OK, great.  I think ideally, we'd import the message with a sender that is a bot user created with the name "Simple Poll".  It looks like we can get that name from the "footer" field only, or am I missing something?  What a weird API.

Reply:
@timabbott https://github.com/rheaparekh/slack2zulip-data-converter/blob/master/slack-export/general/2018-02-23.json#L9-L83 From here, I have a feeling that the message format (fields in the json data) would be different for every bot(I'll investigate this and confirm), and we can see that in the link, there is no such "message" that can be mapped. 

All the bot replies would hence have to be differently mapped.

Reply:
Well, I guess there is a unique ID for the bot, `B9F3T9GR5`.  Might be worth looking up whether there's a way to look up that bot with Slack's API and find out that its name is "Simple Poll".  That field (the name) is part of the metadata for how Slack displays the message, and should be something we can display nicely.